app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,239 @@
+    MyDataPrepApi.createRecipe,
+    apiParams.createRecipeParams,
+    {
+      recipeName: recipeFormData.recipeName,
+      description: recipeFormData.description,
+      directives: recipeSteps,
+    }
+  );
+
+  // This useEffect is for handling getRecipeByName API call response and error
+
+  useEffect(() => {
+    if (recipeByNameResponse) {
+      updateRecipeNameErrorData(
+        {
+          isRecipeNameError: true,
+          recipeNameErrorMessage: T.translate(`${PREFIX}.sameNameErrorMessage`).toString(),
+        },
+        recipeFormData
+      );
+    }
+  }, [recipeByNameResponse]);
+
+  useEffect(() => {
+    if (recipeByNameError) {
+      if (recipeByNameError.statusCode === 404) {
+        updateRecipeNameErrorData(noErrorState, recipeFormData);
+      } else {
+        setSnackbar({
+          open: true,
+          isSuccess: false,
+          message: (recipeByNameError.response as Record<string, string>).message,
+        });
+      }
+    }
+  }, [recipeByNameError]);
+
+  // This useEffect is for handling createRecipe API call response and error
+
+  useEffect(() => {
+    if (createRecipeResponse) {
+      setShowRecipeForm(false);
+      setSnackbar({
+        open: true,
+        isSuccess: true,
+        message: `${recipeSteps.length} ${T.translate(`${PREFIX}.recipeSaveSuccessMessage`)}`,
+      });
+    }
+  }, [createRecipeResponse]);
+
+  useEffect(() => {
+    if (createRecipeError) {
+      setShowRecipeForm(false);
+      setSnackbar({
+        open: true,
+        isSuccess: false,
+        message: (createRecipeError.response as Record<string, string>).message,
+      });
+    }
+  }, [createRecipeError]);
+
+  const updateRecipeNameErrorData = (
+    recipeNameError: IRecipeNameErrorData,
+    formData: IRecipeFormData = recipeFormData
+  ) => {
+    setRecipeNameErrorDataState(recipeNameError);
+    handleSaveButtonMode(formData, recipeNameError);
+  };
+
+  const handleSaveButtonMode = (
+    formData: IRecipeFormData = recipeFormData,
+    nameErrorData: IRecipeNameErrorData = recipeNameErrorData
+  ) => {
+    const shouldDisableSaveButton =
+      formData.recipeName === '' ||
+      formData.description === '' ||
+      formData.recipeName?.trim().length === 0 ||
+      formData.description?.trim().length === 0 ||
+      nameErrorData.isRecipeNameError;
+    setIsSaveDisabled(shouldDisableSaveButton);
+  };
+
+  const handleRecipeFormData = (formData: IRecipeFormData) => {
+    setRecipeFormData(formData);
+    handleSaveButtonMode(formData);
+  };
+
+  const onRecipeNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      recipeName: event.target.value,
+    });
+    validateIfRecipeNameExists.current({
+      recipeName: event.target.value,
+      description: recipeFormData.description,
+    });
+  };
+
+  const onRecipeDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
```

Comment:
dont use React.
instead import from 'react'

Reply:
removed React.ChangeEvent
and imported ChangeEvent  from React